sql: "SELECT~~+~++~\xbd(*)" and handling *StarExpr in EvalExpr

```
Expr: (*parser.StarExpr)(0x20849c028)({
 TableName: (*parser.QualifiedName)(<nil>)
})
panic: eval: unsupported expression type: *parser.StarExpr
```

I'm not sure whether

* the above SELECT should actually parse (doesn't for Postgres, obviously)
* a *StarExpr needs to be handled in EvalExpr. It contains a QualifiedName, which generally needs to be evaluated.

#gofuzz

--- a/sql/parser/parse_test.go
+++ b/sql/parser/parse_test.go
@@ -353,6 +353,9 @@ func TestParse2(t *testing.T) {
 		{`SELECT -0.-/*test*/-1`,
 			`SELECT - 0. - - 1`,
 		},
+		// See #1948.
+		{`SELECT~~+~++~bd(*)`,
+			`SELECT ~ ~ + ~ + + ~ bd(*)`},
 	}
 	for _, d := range testData {
 		stmts, err := Parse(d.sql)

